anta/tests/interfaces.py

```diff
@@ -11,14 +11,59 @@
 import re
 
 # Need to keep Dict and List for pydantic in python 3.8
-from typing import Any, Dict, List, Literal
+from typing import Any, Dict, List, Literal, Union
 
-from pydantic import BaseModel, conint
+from pydantic import BaseModel, Field, conint
 
-from anta.custom_types import Interface
+from anta.custom_types import Interface, SpeedInterface
 from anta.decorators import skip_on_platforms
 from anta.models import AntaCommand, AntaTemplate, AntaTest
 from anta.tools.get_value import get_value
+from anta.tools.utils import get_failed_logs
+
+SpeedPattern = r"^((auto)?\s?(\d{1,4}(\.\d{1})?(g)?(-\d{1,2})?)?|force(d)?\s\d{1,4}(g)?)$"
+
+
+def extract_speed_and_lane(input_speed: str) -> tuple[Any, Any]:
+    """
+    This function extracts the speed and lane information from the input string.
+
+    Parameters:
+    input_speed (str): The input string which contains the speed and lane information.
+
+    Returns:
+    tuple[Any, Any]: The extracted speed from the input string, and the extracted lane from the input string.
+                     If no lane information is found, it returns None.
+    """
```

Comment:
Can you add a small example in the docstring and unit test please.

Reply:
updated doc string and added unit tests.